sched/Kconfig

```diff
@@ -282,11 +282,10 @@ config SMP
 		CPU0 IRQ handler use thread0's stack, but thread0 may switch to CPU1, that
 		will caused IRQ handler stack corruption.
 
-if SMP
-
 config SMP_NCPUS
```

Comment:
For consistency to the adopted naming template, we should remove the `SMP_`  prefix, since it is strictly related to SMP.

We should rename the config to just `NCPUS`.

Reply:
Ok, let me do the rename in the next PR.

Reply:
@gustavonihei after discuss in https://github.com/apache/incubator-nuttx/pull/5247, this change just ensure CONFIG_SMP_NCPUS equal 1 in no SMP case to avoid #ifdef/#else/#endif spread in the code base, but don't allow the user to select value > 1. So I think we can keep it as CONFIG_SMP_NCPUS.